test/test_utils/__init__.py

```diff
@@ -242,7 +242,7 @@ def wrapper(*args, **kwargs):
             if fail_expected_failures:
                 return func(*args, **kwargs)
```

Comment:
When fail_expected_failures is set, we aim to behave just like the standard test runner and do nothing to change / interfere with what happens with the test results.